### PR TITLE
Add `watchers` field to solr and reindex existing content.

### PR DESCRIPTION
**Pull request by deiferni**
With this PR we add the field `watchers` to solr so that it can be queried via the `@solrsearch` endpoint and to allow a user to view "watched" content. Currently it is only possible to watch tasks, but the implementation is generic so that new content can be added if desired.

We add the following new functionality:
- `watchers` field in solr and indexers for plone content. there is one generic indexer which returns an empty list and one task (and forwarding) specific indexer which queries for watchers. Only users that have set themselves manually as watchers (i.e. that have the `WATCHER_ROLE`) are kept in this index.
- add `IDisableCSRFProtection` to existing endpoints so that reindex operations are successful
- add an upgrade-step to reindex existing tasks
- also add the `watchers` field to the `@listing` endpoint and adapt docs

Watchers could then be queried as follows: `http://localhost:8080/fd/@solrsearch?q.raw=watchers:jonas.baumann`

For https://4teamwork.atlassian.net/browse/GEVER-29.

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- API change:
  - [x] Documentation is updated
- New functionality:
  - [x] for `task` also works for `forwarding`
- Upgrade steps (changes in profile):
  - [x] Make it deferrable if possible

### CODE REVIEW

File: opengever/activity/center.py [hunk 1]
```diff
@@ -93,17 +93,17 @@ def remove_watchers_from_resource_by_role(self, oguid, role):
             # for more information about this issue.
             self.session.expire(resource)
 
-    def get_watchers(self, oguid):
-        """Returns a read-only tuple of watchers for a given oguid.
+    def get_watchers(self, oguid, role=None):
+        """Returns watchers for a given resouce identified by oguid.
```
Comment (lukasgraf) [inline opengever/activity/center.py:97]: `resouce` -> `resource`

File: opengever/activity/center.py [hunk 2]
```diff
@@ -93,17 +93,17 @@ def remove_watchers_from_resource_by_role(self, oguid, role):
...
+
+        Allow to filter by role to return only watchers of a certain role.
```
Comment (lukasgraf) [inline opengever/activity/center.py:99]: `Allow` -> `Allows`

File: opengever/activity/tests/test_notification_center.py [hunk 3]
```diff
@@ -142,12 +142,45 @@ def test_get_watchers_returns_a_list_of_resource_watchers(self):
...
+        watchers = self.center.get_watchers(
+            Oguid('fd', '123'), role=WATCHER_ROLE)
+        self.assertItemsEqual([peter, marie], watchers)
+
+    def test_get_watchers_returns_empty_list_when_no_watchers_are_present(self):
+        create(Builder('resource').oguid('fd:123'))
+
+        self.assertEquals([], self.center.get_watchers(Oguid('fd', '123')))
+
+    def test_get_watchers_returns_empty_list_when_no_role_does_not_exist(self):
```
Comment (lukasgraf) [inline opengever/activity/tests/test_notification_center.py:173]: -> `...when_role_does_not_exist`

File: opengever/activity/center.py [hunk 4]
```diff
@@ -234,23 +234,41 @@ def _get_oguid_for(self, item):
             return Oguid.for_object(item)
         return item
 
+    def _reindex_object(self, obj_or_oguid):
```
Comment (lukasgraf) [inline opengever/activity/center.py:237]: `_reindex_watchers`? _(To avoid the impression that the whole object is getting reindexed)_

Comment (deiferni) [reply opengever/activity/center.py:237]: i like it, it clarifies what happens in the method. amended 👍 .

**Comment by deiferni** [comment]
_thanks, amended. will wait with merging for UI feedback for now_

**Pull request merged**